CASMINST-6188 Import / Export of BSS Data (#4156)
Cray-HPE/docs-csm: operations/boot_orchestration/Exporting_and_Importing_BSS_Data.md

Reviewer: This new file should also be linked to from `operations/README.md` (in both the boot orchestration section and the backup/restore section).

Author: Added

> # Export and Importing BSS Data

Reviewer: ```suggestion
# Exporting and Importing BSS Data
```

> 

Reviewer: ```suggestion

- [Prerequisites](#prerequisites)
- [Export BSS boot parameters](#export-bss-boot-parameters)
- [Import BSS boot parameters](#import-bss-boot-parameters)
- [Update BSS after IMS import](#update-bss-after-ims-import)

```

## Prerequisites

- Ensure that the `cray` command line interface (CLI) is authenticated and configured to talk to system management services.
  - See [Configure the Cray CLI](../configure_cray_cli.md).
- In order to use the automated procedures, the latest CSM documentation RPM must be installed on the node where the procedure is being performed.
  - See [Check for latest documentation](../../update_product_stream/README.md#check-for-latest-documentation).

> ### Exporting BSS Boot Parameters

Reviewer: ```suggestion
## Export BSS boot parameters
```

> 1. (`ncn-mw#`) Run the following command to create a `json` file with the boot parameters:

Reviewer: ```suggestion
1. (`ncn-mw#`) Create a JSON file with the boot parameters.
```

   ```bash
   cray bss bootparameters list --format json > cray-bss-boot-parameters-dump.json
   ```

> 1. (`ncn-mw#`) Run the following command to create a `json` file with the boot parameters for only the Compute Nodes:

Reviewer: ```suggestion
1. (`ncn-mw#`) Create a JSON file with the boot parameters for only the compute nodes.
```

   ```bash
   xnames=`cray hsm state components list --type Node --role Compute --format json | jq -r '.[] | map(.ID) | join(",")'`
   echo $xnames
   cray bss bootparameters list --name $xnames --format json > cray-bss-compute-boot-parameters-dump.json
   ```

> ### Restore BSS Boot Parameters

Reviewer: ```suggestion
## Restore BSS boot parameters
```

1. Copy the file generated by the export command to the node where the import procedure is being performed.

> 1. (`ncn-mw#`)  Set up token:

Reviewer: ```suggestion
1. (`ncn-mw#`) Set up an API token.
```

   ```bash

>    export TOKEN=$(curl -k -s -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=$(kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d) https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')

Reviewer: ```suggestion
   export TOKEN=$(curl -k -s -S -d grant_type=client_credentials -d client_id=admin-client \
       -d client_secret=$(kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d) \
       https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
```

   ```

> 1. (`ncn-mw#`) Restore Boot Parameters with the following script (`cray-bss-compute-parameters-dump.json` is the name of the file created from the export command):

Reviewer: ```suggestion
1. (`ncn-mw#`) Restore the boot parameters.

   > `cray-bss-compute-parameters-dump.json` is the name of the file created from the export procedure.
```

   ```bash
   /usr/share/doc/csm/scripts/operations/boot_script_service/bss-restore-bootparameters.sh cray-bss-compute-parameters-dump.json
   ```

> ### Update `ids` and `etags` after IMS import

Reviewer: ```suggestion
## Update BSS after IMS import
```

> - **After** running an IMS

Reviewer: ```suggestion
**After** running an IMS
```

[Automated import procedure](../image_management/Exporting_and_Importing_IMS_Data.md#automated-import-procedure),
run the following script to update the BSS boot parameters.
The IMS import script should have generated a file containing the IMS ID and S3 `etag` mappings -- it is displayed near
the end of the script output.

> 1. (`ncn-mw#`)  Set up token:

Reviewer: ```suggestion
1. (`ncn-mw#`) Set up an API token.
```

   ```bash

>    export TOKEN=$(curl -k -s -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=$(kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d) https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')

Reviewer: ```suggestion
   export TOKEN=$(curl -k -s -S -d grant_type=client_credentials -d client_id=admin-client \
       -d client_secret=$(kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d) \
       https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
```

   ```

1. (`ncn-mw#`)  Run this script

> (`/root/ims-import-export-data/ims-id-maps-post-import-12f86451ce7c49d79e345bee42cc8586.json` is the file from the IMS import procedure):

Reviewer: ```suggestion
1. (`ncn-mw#`) Update BSS based on the changes made during the IMS import.

   > `/root/ims-import-export-data/ims-id-maps-post-import-12f86451ce7c49d79e345bee42cc8586.json` is the file from the IMS import procedure.
```
